armadillo/src/main/java/org/molgenis/armadillo/storage/CharacterSeparatedFile.java

```diff
@@ -0,0 +1,207 @@
+package org.molgenis.armadillo.storage;
+
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvValidationException;
+import java.io.*;
+import java.lang.reflect.Array;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaParseException;
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.io.LocalOutputFile;
+import org.molgenis.armadillo.exceptions.FileProcessingException;
+import org.springframework.web.multipart.MultipartFile;
+
+public class CharacterSeparatedFile {
+  static String STRING = "string";
```

Comment:
private properties? I understand you do this for testability? but then you could have side effects. Or is that okay? It is a bit unconventional ... ;-)